src/ray/raylet/dependency_manager.h

```diff
@@ -192,10 +192,10 @@ class DependencyManager : public TaskDependencyManagerInterface {
 
   /// A struct to represent the object dependencies of a task.
   struct TaskDependencies {
-    TaskDependencies(const std::vector<rpc::ObjectReference> &deps)
-        : num_missing_dependencies(deps.size()) {
+    TaskDependencies(const std::vector<rpc::ObjectReference> &deps) {
```

Comment:
Consider changing this to absl::flat_hash_set<ObjectReference> so that duplicate edge cases can't happen by construction.